include/nbl/video/utilities/CComputeBlit.h

```diff
@@ -0,0 +1,900 @@
+
+	struct SNormalizationCacheKey
+	{
+		asset::IImage::E_TYPE imageType;
+		uint32_t alphaBinCount;
+		asset::E_FORMAT outFormat;
+
+		inline bool operator==(const SNormalizationCacheKey& other) const
+		{
+			return (imageType == other.imageType) && (alphaBinCount == other.alphaBinCount) && (outFormat == other.outFormat);
+		}
+	};
+	struct SNormalizationCacheHash
+	{
+		inline size_t operator() (const SNormalizationCacheKey& key) const
+		{
+			return
+				std::hash<decltype(key.imageType)>{}(key.imageType) ^
+				std::hash<decltype(key.alphaBinCount)>{}(key.alphaBinCount) ^
+				std::hash<decltype(key.outFormat)>{}(key.outFormat);
+		}
+	};
+	core::unordered_map<SNormalizationCacheKey, core::smart_refctd_ptr<video::IGPUComputePipeline>, SNormalizationCacheHash> m_normalizationPipelines;
+
+	struct SBlitCacheKey
+	{
+		uint32_t wgSize;
+		asset::IImage::E_TYPE imageType;
+		uint32_t alphaBinCount;
+		asset::E_FORMAT outFormat;
+		uint32_t smemSize;
+		bool coverageAdjustment;
+
+		inline bool operator==(const SBlitCacheKey& other) const
+		{
+			return (wgSize == other.wgSize) && (imageType == other.imageType) && (alphaBinCount == other.alphaBinCount) && (outFormat == other.outFormat)
+				&& (smemSize == other.smemSize) && (coverageAdjustment == other.coverageAdjustment);
+		}
+	};
+	struct SBlitCacheHash
+	{
+		inline size_t operator()(const SBlitCacheKey& key) const
+		{
+			return
+				std::hash<decltype(key.wgSize)>{}(key.wgSize) ^
+				std::hash<decltype(key.imageType)>{}(key.imageType) ^
+				std::hash<decltype(key.alphaBinCount)>{}(key.alphaBinCount) ^
+				std::hash<decltype(key.outFormat)>{}(key.outFormat) ^
+				std::hash<decltype(key.smemSize)>{}(key.smemSize) ^
+				std::hash<decltype(key.coverageAdjustment)>{}(key.coverageAdjustment);
+		}
+	};
+	core::unordered_map<SBlitCacheKey, core::smart_refctd_ptr<video::IGPUComputePipeline>, SBlitCacheHash> m_blitPipelines;
+
+	uint32_t m_availableSharedMemory;
+	core::smart_refctd_ptr<video::ILogicalDevice> m_device;
+
+	core::smart_refctd_ptr<video::IGPUSampler> samplers[video::IGPUSampler::ETC_COUNT][video::IGPUSampler::ETC_COUNT][video::IGPUSampler::ETC_COUNT][video::IGPUSampler::ETBC_COUNT] = { nullptr };
+
+	CComputeBlit(core::smart_refctd_ptr<video::ILogicalDevice>&& logicalDevice) : m_device(std::move(logicalDevice)) {}
+
+	static inline void dispatchHelper(video::IGPUCommandBuffer* cmdbuf, const video::IGPUPipelineLayout* pipelineLayout, const nbl_glsl_blit_parameters_t& pushConstants, const dispatch_info_t& dispatchInfo)
+	{
+		cmdbuf->pushConstants(pipelineLayout, asset::IShader::ESS_COMPUTE, 0u, sizeof(nbl_glsl_blit_parameters_t), &pushConstants);
+		cmdbuf->dispatch(dispatchInfo.wgCount[0], dispatchInfo.wgCount[1], dispatchInfo.wgCount[2]);
+	}
+
+	core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> getDSLayout(const uint32_t descriptorCount, const asset::E_DESCRIPTOR_TYPE* descriptorTypes, video::ILogicalDevice* logicalDevice) const
```

Comment:
name it `create` not `get`, there's no caching going on

Reply:
Resolved.